# 400_Prep_Movie-Industry

## Purpose
The purpose of this data set is to load in our data and prepare it for analysis.<br> We will also create our measure for success here as this dataset will be merged to two other datasets. 

## Datasets

 - input: movies.csv
 - output: movies.csv

# Loading the Data

Before analysing the data we need to load in the dataset. We will inspect the dataset and carry out any cleaning techniques in order to make sure the information is suitable for efficient analysis. 

In [145]:
import os.path
import pandas as pd
import numpy as np
from scipy import stats

In [146]:
# Ensure the file exists
if  not os.path.exists("movies.csv"):
    print("Missing dataset file")

In [147]:
movies = pd.read_csv('movies.csv',  encoding='latin-1') # Loading in the dataset

# Cleaning the Data

The dataset we have here seems to be cleaned already but we will still carry out some checks to ensure the data is correct.

In [148]:
movies.head(5) # Brings up the first 5 rows

budget                                 company country        director  \
0   8000000.0           Columbia Pictures Corporation     USA      Rob Reiner   
1   6000000.0                      Paramount Pictures     USA     John Hughes   
2  15000000.0                      Paramount Pictures     USA      Tony Scott   
3  18500000.0  Twentieth Century Fox Film Corporation     USA   James Cameron   
4   9000000.0                    Walt Disney Pictures     USA  Randal Kleiser   

       genre        gross                      name rating    released  \
0  Adventure   52287414.0               Stand by Me      R  1986-08-22   
1     Comedy   70136369.0  Ferris Bueller's Day Off  PG-13  1986-06-11   
2     Action  179800601.0                   Top Gun     PG  1986-05-16   
3     Action   85160248.0                    Aliens      R  1986-07-18   
4  Adventure   18564613.0   Flight of the Navigator     PG  1986-08-01   

   runtime  score               star   votes         writer  year  
0       89    8.1        Wil Wheaton  299174   Stephen King  1986  
1      103    7.8  Matthew Broderick  264740    John Hughes  1986  
2      110    6.9         Tom Cruise  236909       Jim Cash  1986  
3      137    8.4   Sigourney Weaver  540152  James Cameron  1986  
4       90    6.9        Joey Cramer   36636  Mark H. Baker  1986

We can look at the shape of the dataset and check if any information is missing.

In [149]:
movies.shape

(6820, 15)

In [150]:
movies.isnull().sum() #checking for missing values

budget      0
company     0
country     0
director    0
genre       0
gross       0
name        0
rating      0
released    0
runtime     0
score       0
star        0
votes       0
writer      0
year        0
dtype: int64

We can set the date as the index of the dataset to enable easier analysis later.

In [151]:
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6820 entries, 0 to 6819
Data columns (total 15 columns):
budget      6820 non-null float64
company     6820 non-null object
country     6820 non-null object
director    6820 non-null object
genre       6820 non-null object
gross       6820 non-null float64
name        6820 non-null object
rating      6820 non-null object
released    6820 non-null object
runtime     6820 non-null int64
score       6820 non-null float64
star        6820 non-null object
votes       6820 non-null int64
writer      6820 non-null object
year        6820 non-null int64
dtypes: float64(3), int64(3), object(9)
memory usage: 799.3+ KB


In [152]:
movies['released'] = pd.to_datetime(movies['released'])
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6820 entries, 0 to 6819
Data columns (total 15 columns):
budget      6820 non-null float64
company     6820 non-null object
country     6820 non-null object
director    6820 non-null object
genre       6820 non-null object
gross       6820 non-null float64
name        6820 non-null object
rating      6820 non-null object
released    6820 non-null datetime64[ns]
runtime     6820 non-null int64
score       6820 non-null float64
star        6820 non-null object
votes       6820 non-null int64
writer      6820 non-null object
year        6820 non-null int64
dtypes: datetime64[ns](1), float64(3), int64(3), object(8)
memory usage: 799.3+ KB


In [153]:
movies

budget                                  company    country  \
0      8000000.0            Columbia Pictures Corporation        USA   
1      6000000.0                       Paramount Pictures        USA   
2     15000000.0                       Paramount Pictures        USA   
3     18500000.0   Twentieth Century Fox Film Corporation        USA   
4      9000000.0                     Walt Disney Pictures        USA   
5      6000000.0                                  Hemdale         UK   
6     25000000.0                   Henson Associates (HA)         UK   
7      6000000.0  De Laurentiis Entertainment Group (DEG)        USA   
8      9000000.0                       Paramount Pictures        USA   
9     15000000.0                     SLM Production Group        USA   
10     8800000.0                            Rimfire Films  Australia   
11    16000000.0           Thorn EMI Screen Entertainment         UK   
12     6000000.0   Twentieth Century Fox Film Corporation        USA   
13    25000000.0   Twentieth Century Fox Film Corporation        USA   
14    15000000.0  De Laurentiis Entertainment Group (DEG)        USA   
15    17000000.0       Producers Sales Organization (PSO)        USA   
16    10000000.0  De Laurentiis Entertainment Group (DEG)        USA   
17    25000000.0                      Geffen Company, The        USA   
18     2700000.0    New Century Entertainment Corporation        USA   
19    35000000.0                       Universal Pictures        USA   
20     2000000.0                          New Line Cinema        USA   
21    11000000.0                           Orion Pictures        USA   
22     4700000.0                             Cannon Films        USA   
23    15000000.0                   Jay Weston Productions        USA   
24    25000000.0                       Paramount Pictures        USA   
25           0.0                         TriStar Pictures        USA   
26           0.0                     Neue Constantin Film      Italy   
27           0.0                         TriStar Pictures        USA   
28    25000000.0                       Paramount Pictures        USA   
29     1900000.0                                  Hemdale        USA   
...          ...                                      ...        ...   
6790  26000000.0                     Broad Green Pictures        USA   
6791         0.0                                 AZ Films      Chile   
6792    130000.0                Changing Film Productions        USA   
6793   8000000.0                            Cinelou Films        USA   
6794    900000.0                 Hidden Empire Film Group        USA   
6795         0.0                RabbitBandini Productions        USA   
6796  15000000.0                           Demarest Films        USA   
6797  10000000.0                                BBC Films         UK   
6798  20000000.0                              StudioCanal         UK   
6799  18000000.0                        Gold Circle Films        USA   
6800  17000000.0         Perfect World Pictures (Beijing)        USA   
6801         0.0                                       42         UK   
6802         0.0                           Mandarin Films     France   
6803   5000000.0                                IM Global        USA   
6804         0.0                           Digic Pictures      Japan   
6805         0.0                          Parts and Labor        USA   
6806         0.0              Bing Feng Bao Entertainment        USA   
6807   4000000.0                    Blumhouse Productions        USA   
6808   3000000.0                           Westerly Films    Ireland   
6809   3800000.0                        Sycamore Pictures        USA   
6810         0.0                              StudioCanal         UK   
6811   8500000.0                                CBS Films        USA   
6812         0.0                             Killer Films        USA   
6813         0.0                  Anna Biller Prod

This dataset was clean from the beginning and so after we searched for any issues in the data we found none. Therefore ther was no cleaning required on this dataset.

# Preparing the Data


## Measure of success

We created our measure of success here because this data set will be merged into two seperate datasets and so to get the best measure we calculated it here while we had the most amount of movies in the dataset.

We created our measure of success using the Harmonic mean. The data we used to create this is the average score of the movies and the gross income of the movies.

The first step in creating the harmonic mean was to get our gross and our score into the same numeric range. We did this by creating ranks for our score and our gross. This then got both of these values into the range 1 to 6820, with 1 being the best and 6820 being the worst rank.

In [154]:
#we used the built in .rank method to rank the scores and the gross, stroing the values in new columns.
movies['scoreRank'] = movies.score.rank(axis=0, method='average', numeric_only=None, na_option='keep', ascending=True, pct=False)
movies['grossRank'] = movies.gross.rank(axis=0, method='average', numeric_only=None, na_option='keep', ascending=True, pct=False)

With thins done we could then create our harmonic mean based on the rankings.

In [155]:
scores  = movies[['grossRank', 'scoreRank']]#creating an array of our gross rank and gross score
movies['HarMean'] = scores.apply(stats.hmean, axis=1)#loading in the array and apply the hmean method to the data creating a harmonic mean.

We then nomalised the harmonic mean to the range 0 and 1 as this made the mean more understandle and comparable.

In [156]:
movies['HarMean'] = ((movies['HarMean']-min(movies['HarMean']))/(max(movies['HarMean'])-min(movies['HarMean'])))#nomalising the harmonic with the standard formula.

In [157]:
movies

budget                                  company    country  \
0      8000000.0            Columbia Pictures Corporation        USA   
1      6000000.0                       Paramount Pictures        USA   
2     15000000.0                       Paramount Pictures        USA   
3     18500000.0   Twentieth Century Fox Film Corporation        USA   
4      9000000.0                     Walt Disney Pictures        USA   
5      6000000.0                                  Hemdale         UK   
6     25000000.0                   Henson Associates (HA)         UK   
7      6000000.0  De Laurentiis Entertainment Group (DEG)        USA   
8      9000000.0                       Paramount Pictures        USA   
9     15000000.0                     SLM Production Group        USA   
10     8800000.0                            Rimfire Films  Australia   
11    16000000.0           Thorn EMI Screen Entertainment         UK   
12     6000000.0   Twentieth Century Fox Film Corporation        USA   
13    25000000.0   Twentieth Century Fox Film Corporation        USA   
14    15000000.0  De Laurentiis Entertainment Group (DEG)        USA   
15    17000000.0       Producers Sales Organization (PSO)        USA   
16    10000000.0  De Laurentiis Entertainment Group (DEG)        USA   
17    25000000.0                      Geffen Company, The        USA   
18     2700000.0    New Century Entertainment Corporation        USA   
19    35000000.0                       Universal Pictures        USA   
20     2000000.0                          New Line Cinema        USA   
21    11000000.0                           Orion Pictures        USA   
22     4700000.0                             Cannon Films        USA   
23    15000000.0                   Jay Weston Productions        USA   
24    25000000.0                       Paramount Pictures        USA   
25           0.0                         TriStar Pictures        USA   
26           0.0                     Neue Constantin Film      Italy   
27           0.0                         TriStar Pictures        USA   
28    25000000.0                       Paramount Pictures        USA   
29     1900000.0                                  Hemdale        USA   
...          ...                                      ...        ...   
6790  26000000.0                     Broad Green Pictures        USA   
6791         0.0                                 AZ Films      Chile   
6792    130000.0                Changing Film Productions        USA   
6793   8000000.0                            Cinelou Films        USA   
6794    900000.0                 Hidden Empire Film Group        USA   
6795         0.0                RabbitBandini Productions        USA   
6796  15000000.0                           Demarest Films        USA   
6797  10000000.0                                BBC Films         UK   
6798  20000000.0                              StudioCanal         UK   
6799  18000000.0                        Gold Circle Films        USA   
6800  17000000.0         Perfect World Pictures (Beijing)        USA   
6801         0.0                                       42         UK   
6802         0.0                           Mandarin Films     France   
6803   5000000.0                                IM Global        USA   
6804         0.0                           Digic Pictures      Japan   
6805         0.0                          Parts and Labor        USA   
6806         0.0              Bing Feng Bao Entertainment        USA   
6807   4000000.0                    Blumhouse Productions        USA   
6808   3000000.0                           Westerly Films    Ireland   
6809   3800000.0                        Sycamore Pictures        USA   
6810         0.0                              StudioCanal         UK   
6811   8500000.0                                CBS Films        USA   
6812         0.0                             Killer Films        USA   
6813         0.0                  Anna Biller Prod

With the harmoic mean created our dataset was now prepared and we could begin merging the datasets.

In [158]:
movies.to_pickle('movies.pkl')#stored the data to a pickle file